File: skills/composio-mcp/examples/python/mcp_integration.py
"""
MCP Integration Example - No Provider Packages Needed

This example demonstrates how to use Composio's MCP endpoints directly
without requiring framework-specific provider packages. This approach
works with any MCP-compatible client including Claude Desktop, Cursor,
and custom implementations.

Requirements:
    pip install composio httpx

Environment:
    COMPOSIO_API_KEY=your-composio-key
"""

import os
import json
import asyncio
import logging
from typing import Any, Dict, List, Optional
from dataclasses import dataclass

import httpx

# Configure logging
logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
)
logger = logging.getLogger(__name__)


@dataclass
class MCPTool:
    """Represents an MCP tool definition."""
    name: str
    description: str
    input_schema: Dict[str, Any]


@dataclass
class MCPResponse:
    """Response from MCP tool execution."""
    content: List[Dict[str, Any]]
    is_error: bool = False


class ComposioMCPClient:
    """
    Direct MCP client for Composio without provider packages.
    
    This client implements the MCP protocol directly, allowing integration
    with any system that supports MCP without needing framework-specific
    provider packages.
    
    Example:
        client = ComposioMCPClient(user_id="user_123")
        tools = await client.list_tools()
        result = await client.call_tool("GITHUB_LIST_REPOS", {"username": "octocat"})
    """
    
    MCP_BASE_URL = "https://mcp.composio.dev"
    
    def __init__(
        self,
        user_id: str,
        api_key: Optional[str] = None,
        toolkits: Optional[List[str]] = None,
        base_url: Optional[str] = None
    ):
        """
        Initialize MCP client.
        
        Args:
            user_id: User identifier for the session
            api_key: Composio API key (or set COMPOSIO_API_KEY env var)
            toolkits: Optional list of toolkits to filter
            base_url: Custom MCP base URL
        """
        self.user_id = user_id
        self.api_key = api_key or os.environ.get("COMPOSIO_API_KEY")
        
        if not self.api_key:
            raise ValueError("API key required. Set COMPOSIO_API_KEY or pass api_key parameter.")
        
        self.base_url = (base_url or self.MCP_BASE_URL).rstrip("/")
        self.toolkits = toolkits or []
        
        self._http_client: Optional[httpx.AsyncClient] = None
        self._tools_cache: Optional[List[MCPTool]] = None
    
    @property
    def mcp_url(self) -> str:
        """Get the MCP endpoint URL."""
        url = f"{self.base_url}/mcp/{self.user_id}"
        if self.toolkits:
            url += f"?toolkits={','.join(self.toolkits)}"
        return url
    
    @property
    def mcp_headers(self) -> Dict[str, str]:
        """Get headers for MCP requests."""
        return {
            "Authorization": f"Bearer {self.api_key}",
            "Content-Type": "application/json",
            "Accept": "application/json"
        }
    
    async def _get_client(self) -> httpx.AsyncClient:
        """Get or create HTTP client."""
        if self._http_client is None:
            self._http_client = httpx.AsyncClient(
                headers=self.mcp_headers,
                timeout=30.0
            )
        return self._http_client
    
    async def _send_request(self, method: str, params: Optional[Dict] = None) -> Dict:
        """Send JSON-RPC request to MCP server."""
        client = await self._get_client()
        
        payload = {
            "jsonrpc": "2.0",
            "id": 1,
            "method": method,
            "params": params or {}
        }
        
        logger.debug(f"MCP Request: {method} - {params}")
        
        response = await client.post(self.mcp_url, json=payload)
        response.raise_for_status()
        
        result = response.json()
        
        if "error" in result:
            error = result["error"]
            raise Exception(f"MCP Error: {error.get('message', 'Unknown error')}")
        
        return result.get("result", {})
    
    async def initialize(self) -> Dict[str, Any]:
        """
        Initialize the MCP connection.
        
        Returns:
            Server capabilities and info
        """
        result = await self._send_request("initialize", {
            "protocolVersion": "2024-11-05",
            "capabilities": {
                "roots": {"listChanged": True}
            },
            "clientInfo": {
                "name": "composio-python-client",
                "version": "1.0.0"
            }
        })
        
        logger.info(f"MCP initialized: {result.get('serverInfo', {}).get('name', 'Unknown')}")
        return result
    
    async def list_tools(self, refresh: bool = False) -> List[MCPTool]:
        """
        List available tools from the MCP server.
        
        Args:
            refresh: Force refresh of cached tools
            
        Returns:
            List of available MCPTool objects
        """
        if self._tools_cache is not None and not refresh:
            return self._tools_cache
        
        result = await self._send_request("tools/list")
        
        tools = []
        for tool_data in result.get("tools", []):
            tools.append(MCPTool(
                name=tool_data["name"],
                description=tool_data.get("description", ""),
                input_schema=tool_data.get("inputSchema", {})
            ))
        
        self._tools_cache = tools
        logger.info(f"Listed {len(tools)} tools")
        return tools
    
    async def call_tool(
        self,
        name: str,
        arguments: Optional[Dict[str, Any]] = None
    ) -> MCPResponse:
        """
        Call a tool on the MCP server.
        
        Args:
            name: Tool name (e.g., "GITHUB_STAR_REPO")
            arguments: Tool arguments
            
        Returns:
            MCPResponse with tool execution results
        """
        logger.info(f"Calling tool: {name}")
        
        result = await self._send_request("tools/call", {
            "name": name,
            "arguments": arguments or {}
        })
        
        return MCPResponse(
            content=result.get("content", []),
            is_error=result.get("isError", False)
        )
    
    async def close(self):
        """Close the client and clean up resources."""
        if self._http_client:
            await self._http_client.aclose()
            self._http_client = None
    
    async def __aenter__(self):
        await self.initialize()
        return self
    
    async def __aexit__(self, exc_type, exc_val, exc_tb):
        await self.close()


def get_mcp_config_for_claude(
    user_id: str,
    api_key: Optional[str] = None,
    toolkits: Optional[List[str]] = None
) -> Dict[str, Any]:
    """
    Generate MCP configuration for Claude Desktop or similar clients.
    
    This generates the JSON configuration that can be added to Claude's
    MCP settings file.
    
    Args:
        user_id: User identifier
        api_key: Composio API key
        toolkits: List of toolkits to enable
        
    Returns:
        Configuration dict for claude_desktop_config.json
    """
    api_key = api_key or os.environ.get("COMPOSIO_API_KEY")
    
    url = f"https://mcp.composio.dev/mcp/{user_id}"
    if toolkits:
        url += f"?toolkits={','.join(toolkits)}"
    
    return {
        "mcpServers": {
            "composio": {
                "url": url,
                "headers": {
                    "Authorization": f"Bearer {api_key}"
                }
            }
        }
    }


def get_mcp_config_for_cursor(
    user_id: str,
    api_key: Optional[str] = None,
    toolkits: Optional[List[str]] = None
) -> Dict[str, Any]:
    """
    Generate MCP configuration for Cursor IDE.
    
    Args:
        user_id: User identifier
        api_key: Composio API key
        toolkits: List of toolkits to enable
        
    Returns:
        Configuration dict for Cursor's MCP settings
    """
    api_key = api_key or os.environ.get("COMPOSIO_API_KEY")
    
    url = f"https://mcp.composio.dev/mcp/{user_id}"
    if toolkits:
        url += f"?toolkits={','.join(toolkits)}"
    
    return {
        "mcp": {
            "servers": {
                "composio": {
                    "transport": "sse",
                    "url": url,
                    "headers": {
                        "Authorization": f"Bearer {api_key}"
                    }
                }
            }
        }
    }


class MCPToolExecutor:
    """
    Helper class for executing tools through MCP in a synchronous context.
    
    Useful for scripts that don't need full async support.
    """
    
    def __init__(self, user_id: str, api_key: Optional[str] = None):
        self.user_id = user_id
        self.api_key = api_key
        self._client: Optional[ComposioMCPClient] = None
    
    def _ensure_client(self):
        if self._client is None:
            self._client = ComposioMCPClient(
                user_id=self.user_id,
                api_key=self.api_key
            )
    
    def list_tools(self) -> List[MCPTool]:
        """List available tools (sync wrapper)."""
        self._ensure_client()
        return asyncio.run(self._async_list_tools())
    
    async def _async_list_tools(self) -> List[MCPTool]:
        async with self._client:
            return await self._client.list_tools()
    
    def call_tool(self, name: str, arguments: Dict[str, Any]) -> MCPResponse:
        """Call a tool (sync wrapper)."""
        self._ensure_client()
        return asyncio.run(self._async_call_tool(name, arguments))
    
    async def _async_call_tool(self, name: str, arguments: Dict[str, Any]) -> MCPResponse:
        async with self._client:
            return await self._client.call_tool(name, arguments)


# SSE Stream handling for real-time MCP
class MCPSSEClient:
    """
    MCP client with Server-Sent Events (SSE) support for real-time streaming.
    
    This is useful for long-running tools or when you need progress updates.
    """
    
    def __init__(
        self,
        user_id: str,
        api_key: Optional[str] = None,
        base_url: str = "https://mcp.composio.dev"
    ):
        self.user_id = user_id
        self.api_key = api_key or os.environ.get("COMPOSIO_API_KEY")
        self.base_url = base_url.rstrip("/")
    
    @property
    def sse_url(self) -> str:
        """SSE endpoint URL."""
        return f"{self.base_url}/sse/{self.user_id}"
    
    async def stream_tool_call(
        self,
        name: str,
        arguments: Dict[str, Any]
    ):
        """
        Call a tool with SSE streaming response.
        
        Yields events as they arrive from the server.
        
        Args:
            name: Tool name
            arguments: Tool arguments
            
        Yields:
            Event dictionaries from SSE stream
        """
        async with httpx.AsyncClient() as client:
            async with client.stream(
                "POST",
                self.sse_url,
                json={
                    "jsonrpc": "2.0",
                    "id": 1,
                    "method": "tools/call",
                    "params": {"name": name, "arguments": arguments}
                },
                headers={
                    "Authorization": f"Bearer {self.api_key}",
                    "Accept": "text/event-stream"
                }
            ) as response:
                async for line in response.aiter_lines():
                    if line.startswith("data: "):
                        data = line[6:]
                        if data and data != "[DONE]":
                            yield json.loads(data)


# Example usage
async def main():
    """Demonstrate MCP integration patterns."""
    
    api_key = os.environ.get("COMPOSIO_API_KEY")
    if not api_key:
        print("Warning: COMPOSIO_API_KEY not set")
        print("Set it with: export COMPOSIO_API_KEY=your-key")
        return
    
    user_id = "demo_user_mcp"
    
    print("=" * 60)
    print("Composio MCP Integration Examples")
    print("=" * 60)
    
    # Example 1: Direct MCP client usage
    print("\n--- Example 1: Async MCP Client ---")
    try:
        async with ComposioMCPClient(user_id=user_id, toolkits=["github"]) as client:
            # List available tools
            tools = await client.list_tools()
            print(f"Available tools: {len(tools)}")
            for tool in tools[:5]:  # Show first 5
                print(f"  - {tool.name}: {tool.description[:50]}...")
            
            # Call a tool (if you have GitHub connected)
            # result = await client.call_tool("GITHUB_LIST_REPOS", {"username": "octocat"})
            # print(f"Result: {result.content}")
    except Exception as e:
        print(f"Error: {e}")
    
    # Example 2: Generate config for Claude Desktop
    print("\n--- Example 2: Claude Desktop Config ---")
    config = get_mcp_config_for_claude(
        user_id=user_id,
        toolkits=["github", "slack"]
    )
    print(json.dumps(config, indent=2))
    
    # Example 3: Generate config for Cursor
    print("\n--- Example 3: Cursor Config ---")
    config = get_mcp_config_for_cursor(
        user_id=user_id,
        toolkits=["github"]
    )
    print(json.dumps(config, indent=2))
    
    # Example 4: Sync wrapper for simple scripts
    print("\n--- Example 4: Sync Tool Executor ---")
    try:
        executor = MCPToolExecutor(user_id=user_id)
        tools = executor.list_tools()
        print(f"Tools via sync executor: {len(tools)}")
    except Exception as e:
        print(f"Error: {e}")
    
    # Example 5: MCP URL and headers for manual integration
    print("\n--- Example 5: Manual Integration Details ---")
    client = ComposioMCPClient(user_id=user_id, toolkits=["github"])
    print(f"MCP URL: {client.mcp_url}")
    print(f"Headers: {json.dumps({k: '...' if 'auth' in k.lower() else v for k, v in client.mcp_headers.items()}, indent=2)}")
    
    print("\n" + "=" * 60)
    print("MCP integration examples completed!")


if __name__ == "__main__":
    asyncio.run(main())

File: skills/composio-mcp/examples/typescript/mcp-integration.ts
/**
 * MCP Integration with Vercel AI SDK
 * 
 * Demonstrates how to use Composio's MCP (Model Context Protocol) endpoint
 * with the Vercel AI SDK for streaming AI interactions.
 * 
 * @example
 * ```bash
 * # Set environment variables
 * export COMPOSIO_API_KEY="your_composio_key"
 * export OPENAI_API_KEY="your_openai_key"
 * 
 * # Run the example
 * npx ts-node mcp-integration.ts
 * ```
 */

import { Composio, MCPEndpoint, ComposioTool } from '../../composio-client';
import { generateText, streamText, tool, CoreTool } from 'ai';
import { openai } from '@ai-sdk/openai';
import { anthropic } from '@ai-sdk/anthropic';
import { z } from 'zod';

// ============================================================================
// MCP Client Implementation
// ============================================================================

/**
 * Simple MCP client for communicating with Composio's MCP endpoint
 */
class MCPClient {
  constructor(
    private readonly endpoint: MCPEndpoint
  ) {}

  /**
   * List available tools from the MCP server
   */
  async listTools(): Promise<ComposioTool[]> {
    const response = await fetch(`${this.endpoint.url}/tools/list`, {
      method: 'POST',
      headers: this.endpoint.headers,
      body: JSON.stringify({ method: 'tools/list' }),
    });

    if (!response.ok) {
      throw new Error(`MCP request failed: ${response.status}`);
    }

    const data = await response.json();
    return data.tools;
  }

  /**
   * Execute a tool via MCP
   */
  async executeTool(name: string, args: unknown): Promise<unknown> {
    const response = await fetch(`${this.endpoint.url}/tools/call`, {
      method: 'POST',
      headers: this.endpoint.headers,
      body: JSON.stringify({
        method: 'tools/call',
        params: { name, arguments: args },
      }),
    });

    if (!response.ok) {
      throw new Error(`MCP tool execution failed: ${response.status}`);
    }

    const data = await response.json();
    return data.content;
  }
}

// ============================================================================
// Vercel AI SDK Integration
// ============================================================================

/**
 * Convert Composio tools to Vercel AI SDK tool format
 */
function convertToVercelTools(
  composioTools: ComposioTool[],
  mcpClient: MCPClient
): Record<string, CoreTool> {
  const tools: Record<string, CoreTool> = {};

  for (const composioTool of composioTools) {
    // Convert JSON Schema to Zod schema
    const zodSchema = jsonSchemaToZod(composioTool.parameters);

    tools[composioTool.name] = tool({
      description: composioTool.description,
      parameters: zodSchema,
      execute: async (args) => {
        console.log(`🔧 Executing via MCP: ${composioTool.name}`);
        const result = await mcpClient.executeTool(composioTool.name, args);
        return result;
      },
    });
  }

  return tools;
}

/**
 * Convert JSON Schema to Zod schema (simplified)
 * In production, use a library like json-schema-to-zod
 */
function jsonSchemaToZod(schema: ComposioTool['parameters']): z.ZodType {
  const shape: Record<string, z.ZodType> = {};

  for (const [key, prop] of Object.entries(schema.properties)) {
    let zodType: z.ZodType;

    switch (prop.type) {
      case 'string':
        zodType = prop.enum 
          ? z.enum(prop.enum as [string, ...string[]])
          : z.string();
        if (prop.description) {
          zodType = zodType.describe(prop.description);
        }
        break;
      case 'number':
      case 'integer':
        zodType = z.number();
        break;
      case 'boolean':
        zodType = z.boolean();
        break;
      case 'array':
        zodType = z.array(z.any());
        break;
      case 'object':
        zodType = z.record(z.any());
        break;
      default:
        zodType = z.any();
    }

    // Make optional if not required
    if (!schema.required?.includes(key)) {
      zodType = zodType.optional();
    }

    shape[key] = zodType;
  }

  return z.object(shape);
}

// ============================================================================
// Streaming Chat Implementation
// ============================================================================

/**
 * Create a streaming chat interface using Vercel AI SDK
 */
class MCPStreamingChat {
  private composio: Composio;
  private session: Awaited<ReturnType<Composio['create']>> | null = null;
  private mcpClient: MCPClient | null = null;
  private tools: Record<string, CoreTool> = {};

  constructor(
    private readonly userId: string,
    private readonly toolkits: string[]
  ) {
    this.composio = new Composio({ debug: true });
  }

  /**
   * Initialize the MCP connection
   */
  async initialize(): Promise<void> {
    console.log('🔗 Connecting to Composio MCP...');

    // Create session
    this.session = await this.composio.create(this.userId, {
      toolkits: this.toolkits,
    });

    // Get MCP endpoint
    const { mcp } = this.session;
    console.log(`📡 MCP Endpoint: ${mcp.url}`);

    // Create MCP client
    this.mcpClient = new MCPClient(mcp);

    // Fetch and convert tools
    const composioTools = await this.mcpClient.listTools();
    this.tools = convertToVercelTools(composioTools, this.mcpClient);

    console.log(`✅ Connected with ${Object.keys(this.tools).length} tools`);
  }

  /**
   * Generate a complete response (non-streaming)
   */
  async generate(
    prompt: string,
    options?: { model?: 'openai' | 'anthropic' }
  ): Promise<string> {
    const model = options?.model === 'anthropic'
      ? anthropic('claude-3-5-sonnet-20241022')
      : openai('gpt-4-turbo-preview');

    console.log(`\n💬 Prompt: ${prompt}`);
    console.log('⏳ Generating response...\n');

    const { text, toolCalls, toolResults } = await generateText({
      model,
      tools: this.tools,
      maxSteps: 5,
      prompt,
      system: `You are a helpful AI assistant with access to various tools.
Use tools when needed to help accomplish tasks. Be concise and helpful.`,
    });

    // Log tool usage
    if (toolCalls && toolCalls.length > 0) {
      console.log(`\n🔧 Tool calls made: ${toolCalls.length}`);
      toolCalls.forEach((call, i) => {
        console.log(`   ${i + 1}. ${call.toolName}`);
      });
    }

    return text;
  }

  /**
   * Stream a response with real-time output
   */
  async stream(
    prompt: string,
    options?: { model?: 'openai' | 'anthropic' }
  ): Promise<void> {
    const model = options?.model === 'anthropic'
      ? anthropic('claude-3-5-sonnet-20241022')
      : openai('gpt-4-turbo-preview');

    console.log(`\n💬 Prompt: ${prompt}`);
    console.log('📤 Streaming response:\n');

    const result = await streamText({
      model,
      tools: this.tools,
      maxSteps: 5,
      prompt,
      system: `You are a helpful AI assistant with access to various tools.
Use tools when needed to help accomplish tasks. Be concise and helpful.`,
      onStepFinish: ({ text, toolCalls, toolResults }) => {
        if (toolCalls && toolCalls.length > 0) {
          console.log('\n🔧 Tool call:');
          toolCalls.forEach(call => {
            console.log(`   - ${call.toolName}: ${JSON.stringify(call.args)}`);
          });
        }
      },
    });

    // Stream the text output
    for await (const textPart of result.textStream) {
      process.stdout.write(textPart);
    }
    console.log('\n');
  }

  /**
   * Multi-turn conversation
   */
  async chat(messages: Array<{ role: 'user' | 'assistant'; content: string }>): Promise<string> {
    const model = openai('gpt-4-turbo-preview');

    const { text } = await generateText({
      model,
      tools: this.tools,
      maxSteps: 5,
      messages: [
        { role: 'system', content: 'You are a helpful assistant with tool access.' },
        ...messages,
      ],
    });

    return text;
  }

  /**
   * Cleanup
   */
  async close(): Promise<void> {
    if (this.session) {
      await this.session.revoke();
      console.log('🔌 MCP session closed');
    }
  }
}

// ============================================================================
// Example Usage
// ============================================================================

/**
 * Basic MCP example with text generation
 */
async function basicMCPExample(): Promise<void> {
  console.log('\n' + '='.repeat(60));
  console.log('📡 Basic MCP Integration Example');
  console.log('='.repeat(60));

  const chat = new MCPStreamingChat('user_mcp_basic', ['github']);

  try {
    await chat.initialize();
    
    const response = await chat.generate(
      'List my GitHub repositories and tell me which one has the most stars.'
    );
    
    console.log('\n🤖 Response:', response);
  } finally {
    await chat.close();
  }
}

/**
 * Streaming example
 */
async function streamingExample(): Promise<void> {
  console.log('\n' + '='.repeat(60));
  console.log('🌊 Streaming MCP Example');
  console.log('='.repeat(60));

  const chat = new MCPStreamingChat('user_mcp_stream', ['github', 'gmail']);

  try {
    await chat.initialize();
    
    await chat.stream(
      'Check my GitHub notifications and summarize any important ones.'
    );
  } finally {
    await chat.close();
  }
}

/**
 * Multi-model example (switching between OpenAI and Anthropic)
 */
async function multiModelExample(): Promise<void> {
  console.log('\n' + '='.repeat(60));
  console.log('🔄 Multi-Model MCP Example');
  console.log('='.repeat(60));

  const chat = new MCPStreamingChat('user_mcp_multi', ['github']);

  try {
    await chat.initialize();
    
    console.log('\n--- Using OpenAI GPT-4 ---');
    const openaiResponse = await chat.generate(
      'What repositories do I have?',
      { model: 'openai' }
    );
    console.log('OpenAI:', openaiResponse);

    console.log('\n--- Using Anthropic Claude ---');
    const claudeResponse = await chat.generate(
      'Summarize my GitHub activity.',
      { model: 'anthropic' }
    );
    console.log('Claude:', claudeResponse);
  } finally {
    await chat.close();
  }
}

/**
 * Direct MCP client usage (without Vercel AI SDK)
 */
async function directMCPExample(): Promise<void> {
  console.log('\n' + '='.repeat(60));
  console.log('🔌 Direct MCP Client Example');
  console.log('='.repeat(60));

  const composio = new Composio();
  const session = await composio.create('user_direct_mcp', {
    toolkits: ['github'],
  });

  // Get MCP endpoint
  const { mcp } = session;
  console.log('MCP URL:', mcp.url);
  console.log('MCP Headers:', Object.keys(mcp.headers));

  // Create client and list tools
  const client = new MCPClient(mcp);
  const tools = await client.listTools();

  console.log('\nAvailable MCP Tools:');
  tools.slice(0, 5).forEach(tool => {
    console.log(`  - ${tool.name}: ${tool.description.slice(0, 50)}...`);
  });

  // Execute a tool directly
  console.log('\nExecuting github_list_repos...');
  try {
    const result = await client.executeTool('github_list_repos', {
      per_page: 5,
    });
    console.log('Result:', JSON.stringify(result, null, 2).slice(0, 200) + '...');
  } catch (error) {
    console.log('Tool execution requires authentication. Connect GitHub first.');
  }

  await session.revoke();
}

// ============================================================================
// Next.js API Route Example
// ============================================================================

/**
 * Example Next.js API route handler for MCP streaming
 * 
 * @example
 * ```typescript
 * // app/api/chat/route.ts
 * import { mcpApiHandler } from './mcp-integration';
 * export const POST = mcpApiHandler;
 * ```
 */
export async function mcpApiHandler(req: Request): Promise<Response> {
  const { messages, userId } = await req.json();

  const composio = new Composio();
  const session = await composio.create(userId || 'anonymous', {
    toolkits: ['github', 'gmail'],
  });

  const mcpClient = new MCPClient(session.mcp);
  const composioTools = await mcpClient.listTools();
  const tools = convertToVercelTools(composioTools, mcpClient);

  const result = await streamText({
    model: openai('gpt-4-turbo-preview'),
    tools,
    maxSteps: 5,
    messages,
  });

  return result.toDataStreamResponse();
}

// ============================================================================
// Main Entry Point
// ============================================================================

async function main(): Promise<void> {
  const args = process.argv.slice(2);
  const mode = args[0] || 'basic';

  console.log('🌐 Composio MCP + Vercel AI SDK Demo\n');

  switch (mode) {
    case 'basic':
      await basicMCPExample();
      break;
    case 'stream':
      await streamingExample();
      break;
    case 'multi':
      await multiModelExample();
      break;
    case 'direct':
      await directMCPExample();
      break;
    default:
      console.log('Usage: npx ts-node mcp-integration.ts [basic|stream|multi|direct]');
      process.exit(1);
  }
}

main().catch(console.error);

export { MCPClient, MCPStreamingChat, convertToVercelTools };
